keps/sig-cloud-provider/20190215-promoting-cloud-provider-labels.md

```diff
@@ -0,0 +1,180 @@
+title: Promoting Cloud Provider Labels to GA
+authors:
+  - @andrewsykim
+owning-sig: sig-cloud-provider
+participating-sigs:
+  - sig-node
+  - sig-storage
+reviewers:
+  - @dims
+  - @liggit
+  - @msau42
+  - @saad-ali
+  - @thockin
+  - TBD
+approvers:
+  - @thockin
+  - @liggit
+editor: TBD
+creation-date: 2019-02-15
+last-updated: 2019-02-15
+status: provisional
+see-also:
+  - "/keps/sig-node/20190130-node-os-arch-labels.md"
+---
+
+# Promoting Cloud Provider Labels to GA
+
+## Table of Contents
+
+   * [Promoting Cloud Provider Labels to GA](#promoting-cloud-provider-labels-to-ga)
+      * [Table of Contents](#table-of-contents)
+      * [Release Signoff Checklist](#release-signoff-checklist)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Test Plan](#test-plan)
+         * [Graduation Criteria](#graduation-criteria)
+            * [Examples](#examples)
+               * [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+               * [Beta -&gt; GA Graduation](#beta---ga-graduation)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Implementation History](#implementation-history)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+
+## Release Signoff Checklist
+
+- [X] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+When node and volume resources are created in Kubernetes, labels may be applied based on the underlying cloud provider of the Kubernetes cluster.
+These labels contain cloud provider information that may be critical to some advanced features (mainly scheduling).
+When these labels were first introduced, they were prefixed with "beta" as the maturity and usage of these labels were not known at the time.
+
+Today, the cloud provider specific labels are:
+* `beta.kubernetes.io/instance-type`
+* `failure-domain.beta.kubernetes.io/zone`
```

Comment:
I believe these labels are already GA'ed:
https://github.com/kubernetes/kubernetes/blob/98104e44f3b70356d944d64e25be3b9f4fc9e021/pkg/kubelet/apis/well_known_labels.go#L35

Reply:
In that particular file they are labeled "GA" (not sure what the history there is), but kubelet + controllers still need to be updated to use the new GA labels.

Reply:
Sure. We can update usages of these labels to the GA versions, but we shouldn't rename them.

Reply:
That constant as far as I can tell is not used anywhere, is it still GA in that case?

Reply:
Unless you mean we shouldn't rename them because beta -> GA should be non-breaking?

Reply:
Considering there seems to already be "failure-domain.kubernetes.io/" labels defined, how do we reconcile them with the "topology.kubernetes.io" labels being proposed here?

Reply:
> "failure-domain.kubernetes.io/" labels defined,

This is interesting, I see them defined but not used anywhere. In that case I think it's safe to remove them?